### PR TITLE
Update DQN Colab example to support export for Barracuda inference

### PR DESCRIPTION
**Pull request by jrupert-unity**
### Proposed change(s)

Updates the GridWorld DQN Colab notebook to take NCHW formatted image as input, as expected by Barracuda and created a wrapper network that provides other essential tensors and ONNX export parameters to ensure compatibility for Barracuda inference.  

Added new scene to the ML-Agent examples Project for GridWorld that removes goal conditioning and leaves all masks as 1 to be compatible with the DQN Colab.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://jira.unity3d.com/browse/MLA-2315

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [x] Other (please describe)

Updated feature.

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Review by maryamhonari — APPROVED** [review]
Thanks, LGTM. Just some nit tweaks.

File: colab/Colab_UnityEnvironment_2_Train.ipynb [hunk 1]
```diff
@@ -1,598 +1,700 @@
...
-   "stem_cell": {
-    "cell_type": "raw",
-    "source": [],
-    "metadata": {
-     "collapsed": false
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "Copy of Colab-UnityEnvironment-2-Train.ipynb",
```
[the remark this answers is not in the document]
Comment (jrupert-unity) [reply colab/Colab_UnityEnvironment_2_Train.ipynb]: Fixed

File: colab/Colab_UnityEnvironment_2_Train.ipynb [hunk 2]
```diff
@@ -1,598 +1,700 @@
...
+        "    # if you have discrete actions ML-agents expects corresponding a mask\n",
+        "    # tensor with the same shape to exist as input\n",
+        "    def forward(self, visual_obs: torch.tensor, mask: torch.tensor):\n",
+        "        qnet_result = self.qnet(visual_obs)\n",
+        "        # Connect mask to keep it from getting pruned\n",
+        "        # Mask values will be 1 if you never call SetActionMask() in\n",
+        "        # WriteDiscreteActionMask() \n",
+        "        qnet_result = torch.mul(qnet_result, mask)\n",
+        "        action = torch.argmax(qnet_result, dim=1, keepdim=True)\n",
+        "        return [action], self.discrete_shape, self.version_number, self.memory_size\n",
```
[the remark this answers is not in the document]
Comment (jrupert-unity) [reply colab/Colab_UnityEnvironment_2_Train.ipynb:668]: Ty, this is important.  Added fix.

File: colab/Colab_UnityEnvironment_2_Train.ipynb [hunk 3]
```diff
@@ -1,598 +1,700 @@
...
+        "### Export PyTorch Model to ONNX"
+      ],
+      "metadata": {
+        "id": "pFEP9JT5KuO-"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "The following cell provides an example of some of the extra things a model needs to work for ML-Agents inference with Barracuda. The GridWorldColab scene is configured to work with this ONNX file.\n",
```
[the remark this answers is not in the document]
Comment (jrupert-unity) [reply colab/Colab_UnityEnvironment_2_Train.ipynb]: Added fixes.

**Review by miguelalonsojr — APPROVED** [review]
LGTM.
